cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -1526,4 +1530,39 @@ mod tests {
     fn build_script_persisted_no_sandbox_on_failure() {
         build_script_persisted("off", false);
     }
+
+    fn assert_derivation_metadata_propagated(keypath: &str, expected: &str, store_path: &Path) {
+        let drv_prop = cmd!("nix", "derivation", "show", store_path)
+            .pipe(cmd!("jq", "-r", "-c", keypath))
+            .read()
+            .unwrap();
+        assert_eq!(drv_prop.as_str(), expected);
+    }
```

Comment:
can we do this without `duct`, `jq` and do we need this at all?

- we have been somewhat comfortably using the command interface and while `cmd!(..).pipe(cmd!(..))` is somewhat more convenient its an entirely different interface.
That said, duct itself seems quite convenient.

- not sure if we really need jq / a generic keypath at all, since were just checking a field , which we could do with serde as well

```rust
let stdout = cmd!("nix", "derivation", "show", store_path).output().unwrap().stdout();
let drv = serde_json::from_bytes(stdout).unwrap()[0];
let version = drv["env"]["version"].as_str().unwrap();
assert_eq!(version, expected);
```

Reply:
besides, i think we can dro[ this test entirely, since apparently we won't be reading the version from the derivation at all

Reply:
I originally had this functionality for testing more than one metadata field. Once we removed the other metadata fields from the derivation I just left this in because it seemed somewhat useful to be able to query derivation metadata.

Reply:
I'm going to leave it in, I've wanted `duct` in a few different situations and I don't think it hurts anything to keep this test around.

Reply:
I don't really have context for this, but I don't think we need to test a behavior we don't really care about. If we're writing a similar test in the future we can always look at git history

If we do keep the test, I have a pretty strong preference for using `serde` over `jq` in Rust unit tests

Reply:
Not sure I understand about not caring about the functionality. We do care about the functionality because we want the version to be set to `unknown` when a user hasn't set their own version. I'm getting a little frustrated that the PR is getting blocked over _adding_ a test.

Also, since the building happens in the Makefile and the Nix builder, this verifies that those properly detect when a version field is or isn't present.

Reply:
> Not sure I understand about not caring about the functionality. We do care about the functionality because we want the version to be set to `unknown` when a user hasn't set their own version.

I don't have context, but that makes sense to me

> I'm getting a little frustrated that the PR is getting blocked over _adding_ a test.

I care about not using jq enough that I'd like to merge https://github.com/flox/flox/pull/2638

I don't care if we merge that before or after this